src/config/routes.rb

```diff
@@ -3,6 +3,7 @@
     resources :functions do
       resources :comments, only: %i[create destroy]
     end
+    resources :saves, only: %i[index create destroy]
```

Comment:
have this under `functions` resource and as a `resource` instead of `resources` for `create` and `destroy` 
and the leave index under the `user` resource